### DISCUSSION TITLE
Serializing properties without groups

### DISCUSSION THREAD

**Discussion by kingschnulli** [api-platform/core#5154] [Q&A]
This maybe related to the serializer, but I'm a bit unsure so I'm asking this question here. Let's say I have an entity Book:

```
/**
 * @ApiResource()
 */
class Book
{
    /**
     * @Groups({"read", "write"})
     */
    public $name;

    /**
     * @Groups("write")
     */
    public $author;

    /**
     * This property has no groups
     */
    public $foo;

    // ...
}
```

What I'm trying to achieve now is to serialize the $foo property, regardless of the current context groups. Is there some way to configure some kind of "default group" for properties where no group was given?

I know I could just add the group to my property, but I was hoping to configure something (or decorate something) so the serializer will return the property $foo.

TLDR: How can I define a default group for all properties?

**Comment by teohhanhui** [comment]
Just define a `default` group yourself? I don't think there's such functionality in the Symfony Serializer.

**Comment by kingschnulli** [comment]
Can you tell me a good place to add this using a decorator or service override? Maybe this is really a more serializer related question

**Comment by teohhanhui** [comment]
I wouldn't suggest adding the `default` group automatically, as it means you lose control on which properties to expose. But in order to always add the `default` group to the serializer context in API Platform, you can decorate the `SerializerContextBuilder`. 😄

**Comment by birkof** [comment]
@teohhanhui How about adding dynamically a serialization group name like `entity.property` on every property of a entity and to use them in ApiResource no/denormalization context?

**Comment by stale** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.
